File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int start = 0, end = nums.length - 1;
        int result = 0;
        while(start <= end){
            int mid = (start + end)/2;
            if(target < nums[0]){
                return 0;
            }
            if(nums[mid] == target){
                return mid;
            } else if(nums[mid] < target){
                start = mid + 1;
                result = mid + 1;
            } else {
                end = mid - 1;
            }
        }
        return result;
    }
}

File: 0035-search-insert-position/0035-search-insert-position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function (nums, target) {
    let low = 0, high = nums.length - 1;
    let ans = nums.length;
    while (low <= high) {
        let mid = Math.floor((low + high) / 2);

        if (nums[mid] >= target) {
            ans = mid;
            high = mid - 1;
        } else {
            low = mid + 1;
        }
    }
    return ans;
};